aiida/engine/processes/calcjobs/tasks.py

```diff
@@ -233,8 +236,8 @@ async def task_retrieve_job(
 
     :raises: TransportTaskException if after the maximum number of retries the transport task still excepted
     """
-    if node.get_state() == CalcJobState.PARSING:
-        logger.warning(f'CalcJob<{node.pk}> already marked as PARSING, skipping task_retrieve_job')
+    if node.get_state() == CalcJobState.DONE:
+        logger.warning(f'CalcJob<{node.pk}> already marked as DONE, skipping task_retrieve_job')
```

Comment:
I'm not entirely sure why of this change. Maybe you already commented on this on one of the meetings we had and I don't remember, but I couldn't find anything in the text of the PR. Why did you add the `DONE` state? And why did you change it here instead of `PARSING`? What is the purpose of these checks here if the states are supposed to be called in order, just a sanity check?

Reply:
> What is the purpose of these checks here if the states are supposed to be called in order, just a sanity check?

No @ramirezfranciscof this is if e.g. the daemon was stopped/restarted during the transport tasks, we don't want to repeat a task that has already been finished.

but yes @sphuber it appears here that you have simply replaced `PARSING` with `DONE`?
Is this the case and what was the reason?
If it is a swap then surely `PARSING` should be removed from `CalcJobState` and also the places in the tests that use `PARSING` should also be replaced with `DONE`.
Technically this would also warrant a database migration, which brings into question if it is really worth it 😬

Reply:
This change was necessary for one of the previous designs where I think the stashing was performed after the parsing and so a new state was necessary. However, in the current design, the stashing (if needed) is done before retrieving and the parsing then follows retrieving as it used to. So indeed now the change is merely replacing `PARSING` with `DONE` so I can revert it. Note that this state attribute is merely for internal purposes and is even removed after the process is terminated, so a migration wouldn't have been necessary.

Reply:
cheers

> Note that this state attribute is merely for internal purposes and is even removed after the process is terminated, so a migration wouldn't have been necessary.

well I meant in the (very rare case) a migration occurred whilst you still had pending calculations in the `PARSING` state, but yeh a technicality to be sure